src/cbc_sdk/workload/compliance_assessment.py

```diff
@@ -0,0 +1,465 @@
+                the action will be executed on multiple devices. Default is None.
+
+        Returns:
+            dict: JSON response containing information about the executed action.
+
+        Raises:
+            ApiError: If the provided action is not one of the allowed actions.
+
+        Example:
+            To reassess an object:
+            benchmark_sets = cb.select(ComplianceBenchmark)
+            benchmark_sets[0].execute_action('REASSESS')
+        """
+        ACTIONS = ('ENABLE', 'DISABLE', 'REASSESS')
+
+        if action.upper() not in ACTIONS:
+            message = (
+                f"\nAction type is required."
+                f"\nAvailable action types: {', '.join(ACTIONS)}"
+                f"\nExample:\nbenchmark_sets = cb.select(ComplianceBenchmark)"
+                f"\nbenchmark_sets[0].execute_action('REASSESS')"
+            )
+            raise ApiError(message)
+
+        args = {"action": action}
+        if device_ids:
+            args['device_ids'] = [device_ids] if isinstance(device_ids, str) else device_ids
+
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/actions"
+        return self._cb.post_object(url, body=args).json()
+
+    def get_benchmark_set_summary(self):
+        """
+        Fetches the compliance summary for the current benchmark set.
+
+        This method constructs the URL for the compliance summary of the benchmark set associated with the current instance,
+        fetches the summary data using the Carbon Black API, and yields Compliance objects for each item in the summary.
+
+        Returns:
+            generator of ComplianceBenchmark: A generator yielding Compliance objects, each representing a summary item.
+        """
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/compliance/summary"
+        results = self._cb.get_object(url)
+        for item in results:
+            yield ComplianceBenchmark(self._cb, initial_data=item)
+
+
+class ComplianceBenchmarkQuery(BaseQuery, CriteriaBuilderSupportMixin,
+                               IterableQueryMixin, AsyncQueryMixin):
+    """
+    A class representing a query for Compliance Benchmark.
+    """
+
+    def __init__(self, doc_class, cb):
+        """
+        Initialize a ComplianceBenchmarkQuery instance.
+
+        Args:
+            doc_class (class): The document class for this query.
+            cb (CBCloudAPI): An instance of CBCloudAPI.
+
+        Returns:
+            ComplianceBenchmarkQuery: An instance of ComplianceBenchmarkQuery.
+        """
+        self._doc_class = doc_class
+        self._cb = cb
+        self._count_valid = False
+        super(BaseQuery, self).__init__()
+
+        self._query_builder = QueryBuilder()
+        self._criteria = {}
+        self._sortcriteria = {}
+        self._total_results = 0
+
+        self._benchmark_set_id = None
+        self._rule_id = None
+
+    def add_criteria(self, key, value, operator='EQUALS'):
```

Comment:
I don't think CIS benchmarks uses operator so we can use the functions from https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/90794cdfdfc69dfc4f6f87989e89da3b3522caf7/src/cbc_sdk/base.py#L1638